aerogear/unifiedpush-operator
# Improve status block of UnifiedPushServer CRD

JIRA: https://issues.jboss.org/browse/AEROGEAR-9958


## Progress

- [x] Updated status in CRD
- [x] Added manageError function to set status & fire event
- [x] Add secondary resources to map in CR status
- [x] Set "ready" status appropriately

## Checklist:


## Verification steps

I've got this all already deployed on an OSD 4 dev cluster, so if you'd like to check it out there, let me know, it might save some time :smiley: 

### Upgrade
- [ ] Deploy operator & UPS from master:
```
git checkout master
make install
# wait for everything to become ready
```
- [ ] Upgrade to configuration from this PR
```
git checkout AEROGEAR-9958
sed -i -e "s|quay.io/aerogear/unifiedpush-operator:.*|quay.io/aerogear/unifiedpush-operator:PR-84|g" deploy/operator.yaml
make install
# ^re-applies all CRDs so they've got the new stuff
```

- [ ] Make sure there are no persistent/blocking errors in the operator logs

### Fresh install
- [ ] Delete the UPS instance:
```
kubectl delete ups --all
```

- [ ] Create a new UPS
```
kubectl apply -f deploy/crds/push_v1alpha1_unifiedpushserver_cr.yaml
```

- [ ] Make sure everything comes up as expected, and there aren't any persistent/blocking errors in the operator logs
